packages/core/test/features/st-import.spec.ts

```diff
@@ -0,0 +1,873 @@
+                        skipLocationCheck: true,
+                    },
+                    {
+                        file: `/entry.st.css`,
+                        message: STImport.diagnostics.ST_IMPORT_EMPTY_FROM(),
+                        severity: `error`,
+                        skipLocationCheck: true,
+                    },
+                ],
+                { partial: true }
+            );
+        });
+        it.skip('should warn on default import with a lowercase first letter', () => {
+            // ToDo: copy functionality
+            expectAnalyzeDiagnostics(
+                `
+                @st-import |comp| from "./file.st.css";
+            `,
+                [
+                    {
+                        message: STImport.diagnostics.DEFAULT_IMPORT_IS_LOWER_CASE(),
+                        file: 'main.css',
+                    },
+                ]
+            );
+        });
+        it('should warn on invalid custom property rename', () => {
+            expectAnalyzeDiagnostics(
+                `
+                |@st-import [--x as z] from "./a.st.css";|
+            `,
+                [
+                    {
+                        message: STImport.diagnostics.INVALID_CUSTOM_PROPERTY_AS_VALUE(`--x`, `z`),
+                        severity: `warning`,
+                        file: `main.st.css`,
+                    },
+                ]
+            );
+        });
+        describe(`syntax`, () => {
+            it('should not allow * import', () => {
+                expectAnalyzeDiagnostics(`|@st-import * as X from "./some/path"|;`, [
+                    {
+                        file: `/entry.st.css`,
+                        message: STImport.diagnostics.ST_IMPORT_STAR(),
+                        severity: `error`,
+                    },
+                ]);
+            });
+            it('should warn on invalid format', () => {
+                expectAnalyzeDiagnostics(
+                    `
+                    |@st-import %# from ("")|;
+                    @st-import f rom "x";
+                `,
+                    [
+                        {
+                            file: `/entry.st.css`,
+                            message: STImport.diagnostics.INVALID_ST_IMPORT_FORMAT([
+                                `invalid missing source`,
+                            ]),
+                            severity: `error`,
+                        },
+                        {
+                            file: `/entry.st.css`,
+                            message: STImport.diagnostics.INVALID_ST_IMPORT_FORMAT([
+                                `invalid missing from`,
+                                `invalid missing source`,
+                            ]),
+                            severity: `error`,
+                            skipLocationCheck: true,
+                        },
+                    ],
+                    { partial: true }
```

Comment:
why partial / support skip location check in `expectAnalyzeDiagnostics`

Reply:
added support for `skipLocationCheck` in non partial `expectAnalyzeDiagnostics` and removed the partial from the test